apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/source/BearerToken.java

```diff
@@ -0,0 +1,48 @@
+package gov.cms.bfd.pipeline.rda.grpc.source;
+
+import static io.grpc.Metadata.ASCII_STRING_MARSHALLER;
+
+import com.google.common.base.Strings;
+import io.grpc.CallCredentials;
+import io.grpc.Metadata;
+import io.grpc.Status;
+import java.util.concurrent.Executor;
+
+/**
+ * A CallCredentials implementation that adds a bearer token to every request. Used to invoke the
+ * RDA API with a JWT token for authorization.
+ */
+public class BearerToken extends CallCredentials {
+  private static final String HEADER_PREFIX = "Bearer ";
+  private static final Metadata.Key<String> HEADER_KEY =
+      Metadata.Key.of("Authorization", ASCII_STRING_MARSHALLER);
+
+  private final String headerValue;
+
+  public BearerToken(String headerValue) {
+    if (Strings.isNullOrEmpty(headerValue)) {
+      throw new IllegalArgumentException("Token value was null or empty.");
+    }
+    this.headerValue = HEADER_PREFIX + headerValue;
```

Comment:
This goes beyond this PR, and I'd like to open a new issue for any work on this.  But the JWT we have, at least for the dev environment, will expire in about a year.  It would be nice to catch this and get a new one from MPSM before it expires.  Any thoughts on the best way to alert the team that the token is about to expire?

Reply:
I created an JIRA issue to track this future enhancement; DCGEO-321